### PR TITLE
chore: added new relic telemetry for spans

### LINKED ISSUE #34072: [Task]: Debug why even after tenant caching implementation there is not much performance boost
### SubTasks

Ref: https://theappsmith.slack.com/archives/C024GUDM0LT/p1717665316665849?thread_ts=1713933688.987279&cid=C024GUDM0LT

### PR DESCRIPTION
**Pull request by NilanshBansal**
Fixes #34072  

## Automation

/ok-to-test tags="@tag.Sanity"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9511509976>
> Commit: d4dc99e250372ce86847b6904c0e642c636ec9ad
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9511509976&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced tenant fetching with improved logging and metrics using Micrometer.

- **Improvements**
	- Added monitoring capabilities to various tenant-related services for better observability and debugging.

### CODE REVIEW

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/constants/spans/TenantSpan.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.appsmith.external.constants.spans;
+
+import static com.appsmith.external.constants.spans.BaseSpan.APPSMITH_SPAN_PREFIX;
+
+public class TenantSpan {
+    public static final String FETCHING_DEFAULT_TENANT = APPSMITH_SPAN_PREFIX + "fetch_default_tenant";
+    public static final String FETCH_TENANT_CACHE_POST_ERROR = APPSMITH_SPAN_PREFIX + "fetch_tenant_cache_error";
+    public static final String FETCH_TENANT_FROM_DB = APPSMITH_SPAN_PREFIX + "fetch_tenant_by_id";
```
Comment (rajatagrawal) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/constants/spans/TenantSpan.java:8]: Can we please rename the value of the constant to be "fetch_tenant_from_db" as well please ?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CacheableRepositoryHelperCEImpl.java [hunk 2]
```diff
@@ -182,13 +188,17 @@ public Mono<Tenant> fetchDefaultTenant(String tenantId) {
...
         Query query = new Query();
         query.addCriteria(andCriteria);
-
-        return mongoOperations.findOne(query, Tenant.class).map(tenant -> {
-            if (tenant.getTenantConfiguration() == null) {
-                tenant.setTenantConfiguration(new TenantConfiguration());
-            }
-            return tenant;
-        });
+        log.info("FETCHING TENANT FROM DATABASE AS NOT PRESENT IN CACHE!");
```
Comment (rajatagrawal) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CacheableRepositoryHelperCEImpl.java:191]: Is this log intentional ?

Comment (rajatagrawal) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CacheableRepositoryHelperCEImpl.java:191]: Is this log intentional ?

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/constants/spans/TenantSpan.java (hunk 1 above)
Comment (NilanshBansal) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/constants/spans/TenantSpan.java]: Good catch! Thanks!

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CacheableRepositoryHelperCEImpl.java (hunk 2 above)
Comment (NilanshBansal) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CacheableRepositoryHelperCEImpl.java:191]: Yes @rajatagrawal this is intentional to be logged to mezmo.

**Review by rajatagrawal — APPROVED** [review]

**Pull request merged**